tests/roots/test-linkcheck-connection-error-on-http-head/index.rst

```diff
@@ -0,0 +1 @@
+`This <https://localhost:7777>`_ tests bug #9306.
```

Comment:
No need for an extra test case, can use the `test-linkcheck-localserver` document root.